atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -240,7 +237,7 @@ public Transaction delegate() {
             try {
                 result = runTaskThrowOnConflict(wrappedTask, tx);
             } finally {
-                lockWatchEventCache.removeTransactionStateFromCache(getTimestamp());
+                lockWatchManager.removeTransactionStateFromCache(getTimestamp());
```

Comment:
It's worth noting a discrepancy in the implementation that we might need to be careful of: LWEC requires this to be called to clear state, whereas LWVC will clear the state on commit (and so this is only for use in an exceptional context, although it doesn't matter if it is called also). Just need to make sure that we definitely do the commit method before this, although I expect we probably do.

Reply:
The only time we do not do the commit method before is if the transaction fails, and we will never do it